src/main/java/com/box/sdk/BoxTaskAssignment.java

```diff
@@ -0,0 +1,214 @@
+
+import com.eclipsesource.json.JsonObject;
+import com.eclipsesource.json.JsonValue;
+
+/**
+ * Represents a task assignment on Box.
+ */
+public class BoxTaskAssignment extends BoxResource {
+
+    private static final URLTemplate TASK_ASSIGNMENT_URL_TEMPLATE = new URLTemplate("tasks_assignments/%s");
+
+    /**
+     * Constructs a BoxTaskAssignment for a task assignment with a given ID.
+     *
+     * @param api the API connection to be used by the resource.
+     * @param id  the ID of the task assignment.
+     */
+    public BoxTaskAssignment(BoxAPIConnection api, String id) {
+        super(api, id);
+    }
+
+    /**
+     * Deletes this task assignment.
+     */
+    public void delete() {
+        URL url = TASK_ASSIGNMENT_URL_TEMPLATE.build(this.getAPI().getBaseURL(), this.getID());
+        BoxAPIRequest request = new BoxAPIRequest(this.getAPI(), url, "DELETE");
+        BoxAPIResponse response = request.send();
+        response.disconnect();
+    }
+
+    /**
+     * Gets information about this task assignment.
+     * @return info about this task assignment.
+     */
+    public Info getInfo() {
+        URL url = TASK_ASSIGNMENT_URL_TEMPLATE.build(this.getAPI().getBaseURL(), this.getID());
+        BoxAPIRequest request = new BoxAPIRequest(this.getAPI(), url, "GET");
+        BoxJSONResponse response = (BoxJSONResponse) request.send();
+        JsonObject responseJSON = JsonObject.readFrom(response.getJSON());
+        return new Info(responseJSON);
+    }
+
+    /**
+     * Updates the information about this task assignment with any info fields that have been modified locally.
+     *
+     * <p>The only fields that will be updated are the ones that have been modified locally. For example, the following
+     * code won't update any information (or even send a network request) since none of the info's fields were
+     * changed:</p>
+     *
+     * <pre>BoxTaskAssignment taskAssignment = new BoxTaskAssignment(api, id);
+     *BoxTaskAssignment.Info info = taskAssignment.getInfo();
+     *taskAssignment.updateInfo(info);</pre>
+     *
+     * @param info the updated info.
+     */
+    public void updateInfo(BoxTask.Info info) {
+        URL url = TASK_ASSIGNMENT_URL_TEMPLATE.build(this.getAPI().getBaseURL(), this.getID());
+        BoxJSONRequest request = new BoxJSONRequest(this.getAPI(), url, "PUT");
+        request.setBody(info.getPendingChanges());
+        BoxJSONResponse response = (BoxJSONResponse) request.send();
+        JsonObject jsonObject = JsonObject.readFrom(response.getJSON());
+        info.update(jsonObject);
+    }
+
+    /**
+     * Contains information about a task assignment.
+     */
+    public class Info extends BoxResource.Info {
+        private BoxItem.Info item;
+        private BoxUser.Info assignedTo;
+        private String message;
+        private Date completedAt;
+        private Date assignedAt;
+        private Date remindedAt;
+        private String resolutionState;
```

Comment:
Looks good except for some minor code style checks. And two minor comments to convert the type of couple of class members from String to Enum.

Also, it would be great if you can 'rebase -i' to squash the commits together.